sys/mips/include/vmparam.h

```diff
@@ -98,9 +98,13 @@
  * offset is calculated.
  */
 #define	SHAREDPAGE		(VM_MAXUSER_ADDRESS - PAGE_SIZE)
-#define	USRSTACK		SHAREDPAGE
+/*
+ * To ensure that the stack base address that is sufficiently aligned to create
+ * a bounded capability we must round down by 16 pages to get to 0x7ffbff0000.
+ */
+#define	USRSTACK		(SHAREDPAGE - (15 * PAGE_SIZE))
 #ifdef __mips_n64
-#define	FREEBSD32_SHAREDPAGE	(((vm_offset_t)0x80000000) - PAGE_SIZE)
+#define	FREEBSD32_SHAREDPAGE	(((vm_offset_t)0x80000000) - (15 * PAGE_SIZE))
```

Comment:
Probably not necessary to change anything about FREEBSD32?  If it is, maybe change its USRSTACK not SHAREDPAGE?

Reply:
FREEBSD32 CheriABI is certainly a long way from something that could work so I'd be tempted to leave it alone.